### ISSUE TITLE
data-uri break the process with no error

### ISSUE THREAD

**Issue by sarvaje** [node-gfx/node-canvas-prebuilt#52]
When you run this code, the process break without any error.

```js
const datauri = 'data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==';
const Canvas = require('canvas-prebuilt');

const image = new Canvas.Image();

image.onerror = function (e) {
    console.log(e);
}
console.log('before');
image.src = datauri;
console.log('after');
```

Expected output:
```bash
before
Error: error while reading from input stream
after
```

Current output:
```
before
```

If I change to `canvas` everything works as expected (any 1.6.x version).

It works with version 2.0.0-alpha.13 but I can't use it because it is use in `jsdom` and they don't support version 2 yet.

Maybe this will be fixed after https://github.com/node-gfx/node-canvas-prebuilt/issues/51 is done.

**Comment by chearon** [comment]
> If I change to canvas everything works as expected (any 1.6.x version).

Are you sure? I don't think that was implemented until Automattic/node-canvas#1193. `canvas@next` or `canvas-prebuilt@next`

**Comment by sarvaje** [comment]
This is working fine with version 1.6.11.

But just for the record:

> Are you sure? I don't think that was implemented until Automattic/node-canvas#1193. canvas@next or canvas-prebuilt@next

Sorry, I though that `src` was supported in all the versions, but in fact I was using this code:
```js
const dataURL = require("data-urls");
const data = dataURL('data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==');
const Canvas = require('canvas-prebuilt');

const image = new Canvas.Image();

image.onerror = function (e) {
    console.log(e);
}
console.log('before');
image.source = data.body;
console.log('after');
```

I was expecting an error, but the script finished suddenly in `image.source = data.body`.

But as I said, this is an issue from the past :)

Thanks!!

**Comment by chearon** [comment]
Oops, you're right, I was thinking of http-style URLs. Glad that fixed it for you.
